hubblestack/extmods/grains/cloud_details.py

```diff
@@ -40,17 +43,27 @@ def _get_aws_details():
         token_request = requests.put(token_url, headers=ttl_header, timeout=3, proxies=proxies)
         token = token_request.text
         aws_token_header = {'X-aws-ec2-metadata-token': token}
-        res = requests.get('http://169.254.169.254/latest/dynamic/instance-identity/document',
-                            headers=aws_token_header, timeout=3, proxies=proxies).json()
-        aws['cloud_account_id'] = res.get('accountId', 'unknown')
-
-        # AWS account id is always an integer number
-        # So if it's an aws machine it must be a valid integer number
-        # Else it will throw an Exception
-        int(aws['cloud_account_id'])
-        aws['cloud_instance_id'] = requests.get('http://169.254.169.254/latest/meta-data/instance-id',
-                                                headers=aws_token_header, timeout=3,
-                                                proxies=proxies).text
+        response = requests.get('http://169.254.169.254/latest/dynamic/instance-identity/document',
```

Comment:
For these kind of calls, retry always helps.
Does retry impact performance much? (Retry for non-200 response code)